src/sidebar/search/AddressInput.tsx

```diff
@@ -84,27 +125,30 @@ class Geocoder {
     }
 
     request(query: string) {
-        const currentId = this.getNextId()
+        this.requestAsync(query).then(() => {})
+    }
 
+    async requestAsync(query: string) {
+        const currentId = this.getNextId()
+        this.timeout.cancel()
         if (!query) return
 
-        this.api
-            .geocode(query)
-            .then(result => {
-                const hits = Geocoder.filterDuplicates(result.hits)
-                if (currentId === this.requestId) this.onSuccess(hits)
-            })
-            .catch(reason => {
-                throw Error('Could not get geocoding results because: ' + reason)
-            })
+        await this.timeout.wait()
+        try {
+            const result = await this.api.geocode(query)
+            const hits = Geocoder.filterDuplicates(result.hits)
+            if (currentId === this.requestId) this.onSuccess(hits)
+        } catch (reason) {
+            throw Error('Could not get geocoding results because: ' + reason)
+        }
     }
 
     private getNextId() {
         this.requestId++
         return this.requestId
     }
 
-    private static filterDuplicates = function(hits: GeocodingHit[]) {
+    private static filterDuplicates = function (hits: GeocodingHit[]) {
```

Comment:
... same here

Reply:
Yes, this is weird.

Reply:
This also seems to be different for IntelliJ 'Reformat Code' and 'Reformat with Prettier'. The first removes the space and the latter adds it...

What is your workflow for this? Normally I sometimes press 'Alt+Shift+L' to reformat the code in case I want it to be formatted, but then I am just using the 'Reformat Code' checkbox in the Git commit Dialog to make sure whatever I am committing is formatted correctly. I could easily remap the shortcut to the 'Reformat with Prettier' action, but unfortunately there is no equivalent before commit action in the commit dialog. So I would have to setup some kind of external pre-commit logic to achieve the same effect. Or maybe tweak the IntelliJ reformat code rules to match the prettier style (but that seems rather fragile).

Reply:
I have set prettier to reformat the code on safe. So basically every file which I'm about to commit is already properly formatted. 

What I'm missing though is automated import optimization. This is something the commit dialog also does for me, but with prettier reformatting on save things get tricky because intellij removes unused imports before the commit, then saves, then prettier kicks in and then I get a warning, that the files I'm about to commit have changed...

The way I have it now, is that I don't do any auto-reformating on commit, but only run code inspection. IntelliJ warns me about unused imports and I remove them manually. It doesn't happen too frequently, but I find it still anoying that I have to take care of imports.

Reply:
Ok thanks, I'll try this as well then. We probably need to re-format the entire code base again then, because when I did this I used 'Reformat code' instead of 'Reformat with Prettier' because I did not know they weren't the same... Maybe this is also why @karussell saw different formatting when using prettier in vscode...